google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -61,32 +63,42 @@ public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguratio
       return;
     }
 
-    AppEngineHelper appEngineHelper = new CloudSdkAppEngineHelper(task.getProject());
-
     AppEngineDeploymentConfiguration deploymentConfig = task.getConfiguration();
 
-    final CancellableRunnable deployRunner =  appEngineHelper.createDeployRunner(
-        logManager.getMainLoggingHandler(),
-        task.getSource(),
-        deploymentConfig,
-        callback);
+    AppEngineEnvironment environment =
+        AppEngineEnvironment.valueOf(deploymentConfig.getEnvironment());
+    if (environment == AppEngineEnvironment.APP_ENGINE_FLEX
+        && Strings.isNullOrEmpty(deploymentConfig.getYamlPath())) {
+      callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml")
+          + "\n" + GctBundle.getString("appengine.deployment.error.suggestflexfacet"));
+      return;
+    }
 
-    if (deployRunner != null) {
-      // keep track of any active deployments
-      synchronized (createdDeployments) {
-        createdDeployments.add(deployRunner);
-      }
+    AppEngineHelper appEngineHelper = new CloudSdkAppEngineHelper(task.getProject());
 
-      ProgressManager.getInstance()
-          .run(new Task.Backgroundable(task.getProject(), GctBundle.message(
-              "appengine.deployment.status.deploying"), true,
-              null) {
-            @Override
-            public void run(@NotNull ProgressIndicator indicator) {
-              ApplicationManager.getApplication().invokeLater(deployRunner);
-            }
-          });
-    }
+    appEngineHelper
+        .createDeployRunner(
+            logManager.getMainLoggingHandler(), task.getSource(), deploymentConfig, callback)
+        .ifPresent(
+            deployRunner -> {
```

Comment:
FWIW Reformatting using Reformat with google-java-format produces the same code.